Assignment4/aajodar-22A243/Task07.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# **Task 07: Querying RDF(s)**
+
+# In[2]:
+
+
+get_ipython().system('pip install rdflib')
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+
+# First let's read the RDF file
+
+# In[3]:
+
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[8]:
+
+
+# RDFLib
+ns = Namespace("http://somewhere#")    
+
+# Visualize the results
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    print(s)
+
+
+# In[9]:
+
+
+# SPARQL
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT ?s WHERE { 
+    ?s RDFS:subClassOf ns:LivingThing . 
+  }
+  ''',
+  initNs = {"RDFS":RDFS, "ns": ns}
+)
+
+# Visualize the results
+for r in g.query(q1):
+ print(r.s)
+
+
+# **TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+# 
+
+# In[14]:
+
+
+# RDFLib
+ns = Namespace("http://somewhere#")    
+
+# Individuals of "Person"
+for s,p,o in g.triples((None, RDF.type, ns.Person)):
+    print(s)
+
+# Individuals of the subclases of "Person"
+for subclass,_,_ in g.triples((None, RDFS.subClassOf, ns.Person)):
+    for ind,_,_ in g.triples((None, RDF.type, subclass)):
+        print(ind)
```

Comment:
This only gets you the subclasses of person, not the subclasses of subclasses like the query below.
The one in sparql is correct, this one is missing small additions